lib/S3.js

```diff
@@ -30,22 +30,43 @@ var S3StorageProvider = function(options) {
     bucket: this._options.bucket,
     region: this._options.region
   })
-}
+
+S3StorageProvider.prototype._cb = function(error, res, callback) {
+  var body = '';
+
+  res.on('data', function (chunk) {
+    body += chunk;
```

Comment:
How big are these responses likely to be? The whole thing is stored in memory here, just so we can pass it into the error which might be problematic at large volumes. Would it be sufficient to just act on `res.statusCode` instead of streaming the body too?  Or maybe only streaming the body if the code wasn't 2xx?